.github/workflows/check_rebuild_tempest.yaml

```diff
@@ -0,0 +1,66 @@
+name: Check and Rebuild Tempest Image
+
+on:
+  workflow_call:
+    inputs:
+      branch:
+        description: The branch to run checks on and re-build images from
+        type: string
+        required: true
+
+jobs:
+  check-update:
+    runs-on: ubuntu-latest
+    outputs:
+      rebuild_image: ${{ steps.check.outputs.rebuild_image }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.branch }}
+      - name: Install yq
+        run: |
+          sudo snap install yq
+      - name: Check if a recent snap updates is found
+        id: check
+        run: |
+          # get the snap name and its channel that the ROCK image should track
+          snap=$(yq '.parts.[].stage-snaps.[]' rocks/tempest/rockcraft.yaml)
+          IFS='/' read -r name channel <<< "$snap"
+          echo "snap: $name, channel: $channel"
+
+          # parse date of the last release and reformat it to seconds
+          format='+%s'
+          last_release_date=$(snap info tempest | grep "2023.2/stable" | awk '{ print $3 }')
+          echo "last release date in channel $channel: $last_release_date"
+          last_release_date_in_seconds=$(date -d "$last_release_date" "$format")
+
+          # confirm the validity of the release date
+          if [[ $? != 0 ]]; then
+            echo "$last_release_date is not a valid date."
+          fi
+
+          # get current date and reformat it to seconds
+          cur_date=$(date)
+          echo "current date: $cur_date"
+          cur_date_in_seconds=$(date -d "$cur_date" "$format")
```

Comment:
See above comment. If it's not `date -I`, I am getting error: `date: invalid date ‘Monday, January 15, 2024 PM12:04:21 HKT`

Reply:
I think the issue maybe be with the `HKT` timezone symbol. It works as intended in workflow env (which is in UTC).

Reply:
Didn't expect that the location impact the function (looks more like a bug in `date` program). If it's okay in the runner, then it should be fine.